### PR TITLE
docs: update documentation

### PR DESCRIPTION
**Pull request by bdryanovski**
Update some of the documentation that we have ...

## PR Checklist

Please check if your PR fulfills the following requirements:

* [ ] Tests for the changes have been added (for bug fixes / features)
* [x] Docs have been added / updated (for bug fixes / features)
* [ ] If applicable, have a visual design approval

## PR Type

What kind of change does this PR introduce?

* [ ] Bugfix
* [ ] Feature
* [ ] Code style update (formatting, local variables)
* [ ] Refactoring (no functional changes, no api changes)
* [ ] Build related changes
* [ ] CI related changes
* [x] Documentation content changes
* [ ] clarity.design website / infrastructure changes
* [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

* [ ] Yes
* [x] No

## Other information

Close: #3709

### CODE REVIEW

File: DEVELOPMENT_CONTRIBUTION.md [hunk 1]
```diff
@@ -97,29 +97,30 @@ Open-source for web UI still feels like the Wild West, so we're trying to take a
...
 
 ```
-[<type>] <title>
-<BLANK LINE>
-<detailed commit message>
-<BLANK LINE>
-<reference to closing an issue>
-<BLANK LINE>
+<type>(optional scope): <description>
+
```
Comment (hippee-lee) [inline DEVELOPMENT_CONTRIBUTION.md]: Can we add back the < BLANK LINE >'s so it is explicit.

File: DEVELOPMENT_CONTRIBUTION.md [hunk 2]
```diff
@@ -97,29 +97,30 @@ Open-source for web UI still feels like the Wild West, so we're trying to take a
 For your commit message, please use the following format:
```
Comment (hippee-lee) [inline DEVELOPMENT_CONTRIBUTION.md]: Can we add a sentence about following the Angular Commit Message Guidlines and link to the format?  (https://github.com/angular/angular/blob/22b96b9/CONTRIBUTING.md#commit-message-format)

File: DEVELOPMENT_CONTRIBUTION.md [hunk 3]
```diff
@@ -129,31 +130,37 @@ As you implement your contribution, make sure all work stays on your local topic
...
 
 ### Taking reviews into account
 
-During the review process of your pull request(s), some changes might be requested by Clarity team members. If that happens, add extra commits to your pull request to address these changes. Make sure these new commits are also signed and follow our commit message format.
+During the review process of your pull request(s), some changes might be requested by Clarity team members.
+If that happens, add extra commits to your pull request to address these changes.
+Make sure these new commits are also signed and follow our commit message format.
 
-**Do not amend your commits or squash them**, it will only slow down the review process by forcing reviewers to check every line again. We'll clean up the commit history ourselves when we finally merge your pull request, you don't have to worry about it.
+Please keep an eye on you Pull-Request and try to address the comments if any, as soon as posible.
```
Comment (gnomeontherun) [inline DEVELOPMENT_CONTRIBUTION.md]: Something I've seen recently on a few PRs is confusion about rebasing onto master, and how to squash. Can we write a snippet about that here?

File: DEVELOPMENT_CONTRIBUTION.md (hunk 1 above)
Comment (mcosta74) [reply DEVELOPMENT_CONTRIBUTION.md]: Will be also nice to document how to put BREAKING CHANGE information

Comment (hippee-lee) [reply DEVELOPMENT_CONTRIBUTION.md]: Do you mean more details like this? Or, just linking to the existing info? 
https://github.com/angular/angular/blob/22b96b9/CONTRIBUTING.md#footer

Comment (mcosta74) [reply DEVELOPMENT_CONTRIBUTION.md]: My personal feeling is to have in this document (maybe also copying it from the angular document). BUt also having an external link would be OK

Comment (hippee-lee) [reply DEVELOPMENT_CONTRIBUTION.md]: I think the link would be better. Otherwise we would have to keep the duplicate content in sync with the original.

Comment (mcosta74) [reply DEVELOPMENT_CONTRIBUTION.md]: works for me, and you can anyway add this later if you wanna do something different than the Angular team is doing

File: CODING_GUIDELINES.md [hunk 4]
```diff
@@ -29,7 +29,12 @@ To ensure the highest quality and maintainability for Clarity components, we adh
 ### SCSS
 
 * SCSS variables should use the `clr` prefix and follow the `kebab-case` convention: `$clr-some-variable`.
-* Coming soon...
+* all classes have to start with `clr` prefix: `clr-sr-only`
+* variables must include `!default` so there could be overwritten
```
Comment (gnomeontherun) [inline CODING_GUIDELINES.md]: > so they can be overwritten.

File: CODING_GUIDELINES.md [hunk 5]
```diff
@@ -29,7 +29,12 @@ To ensure the highest quality and maintainability for Clarity components, we adh
 ### SCSS
 
 * SCSS variables should use the `clr` prefix and follow the `kebab-case` convention: `$clr-some-variable`.
-* Coming soon...
+* all classes have to start with `clr` prefix: `clr-sr-only`
```
Comment (gnomeontherun) [inline CODING_GUIDELINES.md]: > all public classes must start with `clr-` prefix

File: FAQ.md [hunk 6]
```diff
@@ -0,0 +1,28 @@
+#### How to run visual tests
```
Comment (gnomeontherun) [inline FAQ.md]: Why add another file? I would prefer to keep these things in fewer files.

Comment (bdryanovski) [reply FAQ.md]: I was not sure where to put that. And there 3 PR from the past two weeks that had an issue with "how to fix failing visual tests". There could be more but that is the one that I found this week

Comment (bdryanovski) [reply FAQ.md]: I'm ok and to remove it .. maybe the only suitable place to move it will be into BUILD.md

Comment (gnomeontherun) [reply FAQ.md]: Its fine then for now, we can always update it later.

**Pull request merged**